frontend/src/components/form/hash-input.jsx

```diff
@@ -0,0 +1,63 @@
+import { createEffect, createSignal } from "solid-js";
+import { styled } from "solid-styled-components";
+import icon from "../../assets/copy-ico.svg";
+
+const HashInput = (props) => {
+  const [value, setValue] = createSignal(
+    props.defaultValue || `djkbfsdjkbfsdj`
+  );
+  const [isValue, setIsValue] = createSignal(Boolean(props.defaultValue));
+
+  createEffect(() => {
+    setIsValue(Boolean(value()));
+  });
+
+  // copy value of field to clickboard
+  const copyToClickboard = () => {
+    navigator.clipboard.writeText(value());
+  };
+
+  // freeze value in input
+  const freezeInput = (e) => {
+    e.target.value = value();
+  };
+
+  return (
+    <Wrapper>
+      <Field
+        type={`text`}
+        placeholder={` `}
+        id={props.name}
+        name={props.name}
+        onChange={freezeInput}
+        value={value()}
+        disabled={!isValue()}
```

Comment:
Please, rename `isValue` to `isEnabled` or `hasValue`

Reply:
done